keras/legacy/interfaces.py

```diff
@@ -141,3 +141,54 @@ def lstm_args_preprocessor(args, kwargs):
     value_conversions={'dim_ordering': {'tf': 'channels_last',
                                         'th': 'channels_first',
                                         'default': None}})
+
+legacy_evaluate_generator_support = generate_legacy_interface(
+    allowed_positional_args=['generator', 'steps', 'max_q_size',
+                             'workers', 'pickle_safe'],
+    conversions=[('val_samples', 'steps'),
+                 ('nb_worker', 'workers')])
+
+
+# For fit_generator, evaluate_generator, predict_generator methods
+def legacy_generator_methods_support(func):
```

Comment:
@fchollet Correct me if I'm wrong, but would the conversion interface for class method be any different than that for the layer constructors (except for the error messages, as you pointed out)